reloc.ml

```diff
@@ -389,8 +389,11 @@ let add_reloc_table obj obj_name p =
         | `x64, 0x08 when not !no_rel_relocs-> 0x0003 (* rel32_4 *)
         | `x64, 0x06 when not !no_rel_relocs-> 0x0005 (* rel32_2 *)
 
-        | `x86, (0x0a (* IMAGE_REL_I386_SECTION *) |
-                 0x0b (* IMAGE_REL_I386_SECREL*) ) ->
+        | `x64, 0x03 (* IMAGE_REL_AMD64_ADDR32NB *) ->
+            0x0100 (*  Win64 EH data structures relocs: ignore *)
```

Comment:
Well, if we don't ignore them, `flexlink` fails to process x64 .obj files that were compiled by C++ compiler. If we ignore them, we just leave them 'as is' to Win32 loader, not breaking anything. Maybe to add some tests to be sure that it really works?

Reply:
> If we ignore them, we just leave them 'as is' to Win32 loader, not breaking anything.

Well, if we ignore them, we are actually removing them from the locations managed by Win32 and we won't patch the location in our loader either.  Meaning that the memory address could be left with a dummy value, possibly resulting in arbitrary behavior when the value is used at runtime.

Reply:
I see. This relocation kind is RVA (relative virtual address), and it's easy to handle it correctly. I will try to do that. Unfortunately flexdll does not contain comprehensive relocation tests to add some test case to. How to demonstrate that this kind really works under flexdll/x64?

Reply:
I would first create a test case that fails at link time with the current version and at runtime when ignoring the relocation, and check that it works fine after adding proper support for the relocation.

What about splitting the PR into multiple ones?  I understand this is more work on your side, but it will simplify discussing and merging rather independent parts.

Reply:
> I would first create a test case that fails at link time with the current version and at runtime when ignoring the relocation, and check that it works fine after adding proper support for the relocation.

Ok

> What about splitting the PR into multiple ones?

I can create separate PRs for all six commits, no problem.